android/src/main/java/com/apptentive/android/sdk/reactlibrary/ApptentiveModule.kt

```diff
@@ -67,20 +68,23 @@ class ApptentiveModule(private val reactContext: ReactApplicationContext) :
     Apptentive.registerApptentiveActivityInfoCallback(this)
 
     Log.d(REACT_NATIVE_TAG, "Observing Message Center Notification")
-    Apptentive.messageCenterNotificationObservable.observe { notification ->
-      Log.v(REACT_NATIVE_TAG, "Message Center notification received: $notification")
-      val eventEmitter =
-        reactContext.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
-      val map = WritableNativeMap()
-      map.putInt("count", notification?.unreadMessageCount ?: 0)
-
-      eventEmitter.emit(UNREAD_MESSAGE_COUNT_CHANGED, map)
-    }
+    Apptentive.messageCenterNotificationObservable.observe(::observeNewMessage)
```

Comment:
ReactNative doesn't observe for SurveyFinished events through `eventNotificationObservable`?

Reply:
Also, There is no control over if the app wants to really listen to the events? Flutter exposes a method called `registerListeners`. Only if this is called from the app the listeners are registered.

Reply:
Not currently.

No, it is always transmitting whether it is listened to or not.

Reply:
In native SDK we expose that API to the app to make its choice. Is there a reason, you chose this way?